Sage/carbon
# test(detail): add accessibility tests

### Proposed behaviour
- Add `accessibility` Cypress tests for the `Detail` component to use the new cypress-component-react framework for testing.

### Current behaviour
Old approach for accessibility tests

### Checklist



#### QA


### Additional context



### Testing instructions

- [] Run `npx cypress open --component` to check if there is newly added test.file
- [] Check if the `detail.test.js` file passed
- [] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `action-popover` tests are not running twice.
